### PR TITLE
[Functions] Mask sensitive data on `store_function`

### PR DESCRIPTION
**Pull request by Tankilevitch**
Phase 2 of - #3076 
Related to https://jira.iguazeng.com/browse/ML-3156

Multiple adjustments had to be done: 
 - On store function masking credentials if `auth_info` is provided (will be done on `store_function` endpoint)
 - Changed the concatenation of the feature store merger to `-` instead of `_` to align with the mlrun function naming conventions.
 - Removed functions from `tests_dbs` and removed all tests of runtime base which are no longer relevant as we are no longer support running runtime base as well as storing unstructured functions.

### CODE REVIEW

**Review by theSaarco — CHANGES_REQUESTED** [review]
Looks good in general. Have some comments and suggestions...

File: tests/integration/sdk_api/projects/test_project.py [hunk 1]
```diff
@@ -54,18 +54,16 @@ def test_overwrite_project(self):
...
+        for label_name, label_value in labels.items():
+            function.set_label(label_name, label_value)
+
+        function_names = ["function-name-1", "function-name-2", "function-name-3"]
         function_tags = ["some-tag", "some-tag2", "some-tag3"]
         for function_name in function_names:
             for function_tag in function_tags:
                 db.store_function(
-                    function,
+                    function.to_dict(),
```
Comment (alonmr) [inline tests/integration/sdk_api/projects/test_project.py:66]: we should probably add a type hint for `function` in `store_function`

File: mlrun/runtimes/pod.py [hunk 2]
```diff
@@ -952,10 +949,11 @@ def get_env(self, name, default=None):
         If it's a scalar value, will return it, if the value is from source, return the k8s struct (V1EnvVarSource)"""
         for env_var in self.spec.env:
             if get_item_name(env_var) == name:
-                value = get_item_name(env_var, "value")
-                if value is not None:
-                    return value
-                return get_item_name(env_var, "value_from")
+                # valueFrom is a workaround for now, for some reason the envs aren't getting sanitized
```
Comment (alonmr) [inline mlrun/runtimes/pod.py:952]: sounds like s bug. shouldn't we resolve this instead of hiding it?

**Review by alonmr — APPROVED** [review]
LGTM. Left a question and a suggestion.

**Review by alonmr — APPROVED** [review]

**Review by theSaarco — APPROVED** [review]
👍

**Pull request merged**